demos/dotnet/dotnet/dotnet.cs

```diff
@@ -18,99 +18,148 @@ static class Program
 
     static void Main(string[] args)
     {
+        // Defaults
         if (args.Length == 0)
         {
-            args = new string[] { "/build" };
+            args = new string[] { "/target:exe" };
         }
 
-        if (args[0] == "/log")
-        {
-            Log.IsEnabled = true;
-        }
+        // Arguments specifications used for parsing
+        var commandFunctions = new List<String>() { "/new", "/clean", "/edit", "/?", "/help" };
+        var commandSwitches = new List<String>() {"/log", "/optimize", "/unsafe"};
+        var commandSwitchesWithSpecifications = new List<String>() { "/target", "/recurse", "/debug" };
+        var targetSpecifications = new List<String>() { "exe", "library" };
+        var debugSpecifications = new List<String>() { "full", "pdbonly" };
```

Comment:
You might want to make these read-only statics, but your call. Otherwise looks good, so please feel free to merge.